dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -239,6 +242,40 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
         return tool;
     }
 
+    /**
+     * Construct the image_name for ImageData
+     *
+     * @return The full image name
+     */
+    private static String constructImageName(final Image image) {
+        DockerSpecifier specifier = image.getSpecifier();
+        Registry registry = image.getImageRegistry();
+        String fullRepositoryName = image.getRepository();
+
+        if (registry != Registry.DOCKER_HUB) {
+            // If the registry is not Docker Hub (ex: Quay), prepend the image registry docker path
+            fullRepositoryName = String.join("/", image.getImageRegistry().getDockerPath(), fullRepositoryName);
+        }
+
+        // Check if specifier is null because tool images don't have the DockerSpecifier set properly yet. For now, if it's null, assume that it's a tag
+        // TODO: Remove null check once the DockerSpecifier for tool images are set properly
+        if (specifier == null || specifier == DockerSpecifier.TAG) {
+            return String.join(":", fullRepositoryName, image.getTag());
+        } else if (specifier == DockerSpecifier.DIGEST) {
+            // The image's sha256 checksum is the image's digest
+            String imageDigest = image.getChecksums().stream()
```

Comment:
FYI, you can also do:

```
Optional<String> imageDigest = image.getChecksums().stream()
.filter(checksum -> checksum.getType().equals("sha256"))
.findFirst();
```
On the downside, that returns an `Optional`, so then you have call `get` on it (or ideally check if it's not empty, but seems like you know this will always succeed?).

Reply:
can also use `orElseThrow`